src/log.c

```diff
@@ -323,11 +323,15 @@ int DEFAULT_SYMVER_PRE(fi_log_enabled)(const struct fi_provider *prov,
 		enum fi_log_subsys subsys)
 {
 	uint64_t flags = 0;
+	int ret = 0;
 
 	if (ofi_prov_ctx(prov)->disable_logging)
 		flags |= FI_LOG_PROV_FILTERED;
 
-	return log_fid.ops->enabled(prov, level, subsys, flags);
+	pthread_mutex_lock(&common_locks.ini_lock);
+	ret = log_fid.ops->enabled(prov, level, subsys, flags);
+	pthread_mutex_unlock(&common_locks.ini_lock);
```

Comment:
This introduces significant overhead to logging, even when it's disabled.  We don't want any of these locks in the logging calls

Reply:
Thanks! I was noticing that so I added the evaluating label. Do you have a different fix suggestion?